src/toxcore/dht/kbucket.rs

```diff
@@ -452,6 +455,10 @@ impl Kbucket {
         let mut bucket = Bucket::new(Some(4));
         for buc in &*self.buckets {
             for node in &*buc.nodes {
+                if !IsGlobal::is_global(&node.saddr.ip()) {
```

Comment:
Actually we have to filter out LAN IPs only in case if the IP of the packet who send to GetNodes is not LAN. If it is LAN we may send him LAN IPs

Reply:
done